sdk/videoanalyzer/video-analyzer-edge/review/video-analyzer-edge.api.md

```diff
@@ -0,0 +1,745 @@
+    samplingOptions?: SamplingOptions;
+};
+
+// @public (undocumented)
+export type ExtensionProcessorBaseUnion = ExtensionProcessorBase | GrpcExtension | HttpExtension;
+
+// @public
+export type FileSink = SinkNodeBase & {
+    "@type": "#Microsoft.VideoAnalyzer.FileSink";
+    baseDirectoryPath: string;
+    fileNamePattern: string;
+    maximumSizeMiB: string;
+};
+
+// @public
+export interface GeneratedClientOptionalParams extends coreHttp.ServiceClientOptions {
+    endpoint?: string;
+}
+
+// @public
+export type GrpcExtension = ExtensionProcessorBase & {
+    "@type": "#Microsoft.VideoAnalyzer.GrpcExtension";
+    dataTransfer: GrpcExtensionDataTransfer;
+    extensionConfiguration?: string;
+};
+
+// @public
+export interface GrpcExtensionDataTransfer {
+    mode: GrpcExtensionDataTransferMode;
+    sharedMemorySizeMiB?: string;
+}
+
+// @public
+export type GrpcExtensionDataTransferMode = string;
+
+// @public
+export type HttpExtension = ExtensionProcessorBase & {
+    "@type": "#Microsoft.VideoAnalyzer.HttpExtension";
+};
+
+// @public
+export type HttpHeaderCredentials = CredentialsBase & {
+    "@type": "#Microsoft.VideoAnalyzer.HttpHeaderCredentials";
+    headerName: string;
+    headerValue: string;
+};
+
+// @public
+export type ImageFormatBmp = ImageFormatProperties & {
+    "@type": "#Microsoft.VideoAnalyzer.ImageFormatBmp";
+};
+
+// @public
+export type ImageFormatJpeg = ImageFormatProperties & {
+    "@type": "#Microsoft.VideoAnalyzer.ImageFormatJpeg";
+    quality?: string;
+};
+
+// @public
+export type ImageFormatPng = ImageFormatProperties & {
+    "@type": "#Microsoft.VideoAnalyzer.ImageFormatPng";
+};
+
+// @public
+export interface ImageFormatProperties {
+    "@type": "#Microsoft.VideoAnalyzer.ImageFormatRaw" | "#Microsoft.VideoAnalyzer.ImageFormatJpeg" | "#Microsoft.VideoAnalyzer.ImageFormatBmp" | "#Microsoft.VideoAnalyzer.ImageFormatPng";
+}
+
+// @public (undocumented)
+export type ImageFormatPropertiesUnion = ImageFormatProperties | ImageFormatRaw | ImageFormatJpeg | ImageFormatBmp | ImageFormatPng;
+
+// @public
+export type ImageFormatRaw = ImageFormatProperties & {
+    "@type": "#Microsoft.VideoAnalyzer.ImageFormatRaw";
+    pixelFormat: ImageFormatRawPixelFormat;
+};
+
+// @public
+export type ImageFormatRawPixelFormat = string;
+
+// @public
+export interface ImageProperties {
+    format?: ImageFormatPropertiesUnion;
+    scale?: ImageScale;
+}
+
+// @public
+export interface ImageScale {
+    height?: string;
+    mode?: ImageScaleMode;
+    width?: string;
+}
+
+// @public
+export type ImageScaleMode = string;
+
+// @public
+export type IotHubMessageSink = SinkNodeBase & {
+    "@type": "#Microsoft.VideoAnalyzer.IotHubMessageSink";
+    hubOutputName: string;
+};
+
+// @public
+export type IotHubMessageSource = SourceNodeBase & {
+    "@type": "#Microsoft.VideoAnalyzer.IotHubMessageSource";
+    hubInputName?: string;
+};
+
+// @public
+export const enum KnownGrpcExtensionDataTransferMode {
```

Comment:
are we making these const or is the generator? If it's the generator, we should make sure to `preserveConstEnums` in your tsconfig so JS customers can use them

Reply:
Yes it is the generator. I've added that option to our tsconfig